File: JumpGame.cpp
#include <vector>

class Solution {
public:
    bool canJump(vector<int>& nums) {
        vector<int> memo;
        for(int i=0;i<nums.size();i++){
            memo.push_back(0);
        }
        memo[0] = 1;
        
        for(int i=0;i<nums.size();i++){
            if (memo[i] == 0){
                i++;
                continue;
            }else{
                for(int j=i;j<i+nums[i]+1;j++){
                    if(j>=nums.size()){
                        break;
                    }
                    memo[j] = 1;
                }
            }
        }
        if (memo[memo.size()-1]){
            return true;
        }else{
            return false;
        }   
    }
};

File: JumpGame.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        mx = 0
        for i in range(len(nums)):
            if i > mx:
                return False
            else:
                mx = max(mx, i+nums[i])
                if mx >= len(nums)-1:
                    return True
        return True
